app/src/main/java/com/rizadwi/snapsift/datasource/service/NewsService.kt

```diff
@@ -0,0 +1,34 @@
+package com.rizadwi.snapsift.datasource.service
+
+import com.rizadwi.snapsift.datasource.service.dto.response.ArticleResponse
+import com.rizadwi.snapsift.datasource.service.dto.response.SourceResponse
+import retrofit2.Response
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface NewsService {
+
+    @GET("top-headlines/sources")
+    suspend fun getNewsSourcesByCategory(
+        @Query("category") category: String
+    ): Response<SourceResponse>
+
+    @GET("top-headlines/sources")
+    suspend fun getNewsSources(): Response<SourceResponse>
+
+    @GET("everything")
+    suspend fun getHeadlineArticles(
+        @Query("q") q: String,
```

Comment:
for multiple query key, try to use query map

Reply:
i'll do the refactoring